src/analysis/locally_convex/with_seminorms.lean

```diff
@@ -283,12 +283,83 @@ lemma with_seminorms.with_seminorms_eq {p : seminorm_family 𝕜 E ι} [t : topo
 variables [topological_space E]
 variables {p : seminorm_family 𝕜 E ι}
 
+lemma with_seminorms.topological_add_group (hp : with_seminorms p) : topological_add_group E :=
+begin
+  rw hp.with_seminorms_eq,
+  exact add_group_filter_basis.is_topological_add_group _
+end
+
 lemma with_seminorms.has_basis (hp : with_seminorms p) : (𝓝 (0 : E)).has_basis
   (λ (s : set E), s ∈ p.basis_sets) id :=
 begin
   rw (congr_fun (congr_arg (@nhds E) hp.1) 0),
   exact add_group_filter_basis.nhds_zero_has_basis _,
 end
+
+lemma with_seminorms.has_basis_zero_ball (hp : with_seminorms p) : (𝓝 (0 : E)).has_basis
+  (λ sr : finset ι × ℝ, 0 < sr.2) (λ sr, (sr.1.sup p).ball 0 sr.2) :=
+begin
+  refine ⟨λ V, _⟩,
```

Comment:
```suggestion
  intro V,
```
or am I missing something?

Reply:
I see where you're coming from, but that does not seem to work: ```intro tactic failed, Pi/let expression expected.```

Reply:
`filter.has_basis` is a one-field structure, so you'd need to do either `split, intro V` or `rw has_basis_iff, intro V`. I like the `refine` version, but take whatever looks more readable to you.

Reply:
Thank you for the clarification! With that in mind, I think the refine version wins out.